core/app/models/spree/product.rb

```diff
@@ -274,6 +274,10 @@ def find_variant_property_rule(option_value_ids)
       end
     end
 
+    def cache_key
+      super + "/prices/" + prices.currently_valid.pluck(:id).join("/")
```

Comment:
I think it would be best to digest the prices. I don't know if we'll encounter an issue with key length if we have thousands of prices, but it's better to err on the side of safety.